### PR TITLE
Enable union validation

### PR DESCRIPTION
**Pull request by gnawf**
Honestly not quite sure why this was commented out… running this on central schema seems fine.

### CODE REVIEW

File: test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ overallSchema:
 # language=GraphQL
 underlyingSchema:
   Issues: |
-    union HasName = Issue | Edible
+    union HasName = Issue | Edible | Troll | User
```
Comment (gnawf) [inline test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml:25]: Honestly not sure why these tests are set up like this… this is not desirable behavior.

File: lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt [hunk 2]
```diff
@@ -474,50 +474,6 @@ class NadelTypeValidationTest : DescribeSpec({
             assert(error.overallField.name == "nextgenSpecific")
         }
 
-        it("ignores extension union members declared by other services") {
```
Comment (felipe-gdr) [inline lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt:477]: Can we add tests for the checks performed by `NadelUnionValidation`?

File: test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml (hunk 1 above)
Comment (felipe-gdr) [reply test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml:25]: Shall we remove them or replace them with new tests using the new format?

File: lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt (hunk 2 above)
Comment (gnawf) [reply lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt:477]: Yeah my bad, got distracted by E2E tests and forgot. Will add some.

File: test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml (hunk 1 above)
Comment (gnawf) [reply test/src/test/resources/fixtures/deep renames/deep-rename-not-asked-for-with-unions.yml:25]: I think I'll just leave them, just not sure why the overall schema had more members to begin with…

File: lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt (hunk 2 above)
Comment (gnawf) [reply lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt:477]: Added

Comment (felipe-gdr) [reply lib/src/test/kotlin/graphql/nadel/validation/NadelTypeValidationTest.kt:477]: Good stuff!!! Thanks for that.

**Review by felipe-gdr — APPROVED** [review]

**Pull request merged**